### PR TITLE
Implement read integer in applicant data

### LINKED ISSUE #491: Implement readInteger in ApplicantData
### Description
Similar to the `readString` method, implement a `readInteger` method that attempts to read an int at the given path in the [`ApplicantData`](https://github.com/seattle-uat/civiform/blob/main/universal-application-tool-0.0.1/app/services/applicant/ApplicantData.java).

### Additional context
Our question system has a number of recognized [`ScalarType`s](https://github.com/seattle-uat/civiform/blob/main/universal-application-tool-0.0.1/app/services/question/ScalarType.java#L6) (as of writing this, it only has `STRING` and `INT`). The `ApplicantData` `read` methods should attempt to parse values in its underlying JSON object into the given scalar type.

### PR DESCRIPTION
**Pull request by fripi89**
### Description
Impelemented `readInteger` in ApplicantData and added test for method

### Checklist
- [x] Created tests which fail without the change

### Issue(s)
Closes #491

### CODE REVIEW

File: universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java [hunk 1]
```diff
@@ -79,6 +79,16 @@ public void readString_findsCorrectValue() throws Exception {
...
+  
+  @Test
+  public void readInteger_findsCorrectValue() throws Exception {
+    String testData = "{ \"applicant\": { \"age\": 30 } }";
+    ApplicantData data = new ApplicantData(testData);
+
+    Optional<Integer> found = data.readInteger(Path.create("applicant.age"));
+
+    assertThat(found).hasValue(30);
+  }
```
Comment (bion) [inline universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java:91]: Suggestion (blocking): please add a test that passes a path to a bad value that isn't present and asserts that it returns an empty optional.

**Review by bion — COMMENTED** [review]
Hi, thanks for this! One comment and then ready for approval :-)

File: universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java [hunk 2]
```diff
@@ -98,4 +117,14 @@ public void readString_returnsEmptyWhenTypeMismatch() {
...
+
+  @Test
+  public void readInteger_returnsEmptyWhenTypeMismatch() {
+    String testData = "{ \"applicant\": { \"object\": { \"name\": \"John\" } } }";
+    ApplicantData data = new ApplicantData(testData);
+
+    Optional<Integer> found = data.readInteger(Path.create("applicant.object.name"));
+
+    assertThat(found).isEmpty();
+  }
```
Comment (bion) [inline universal-application-tool-0.0.1/test/services/applicant/ApplicantDataTest.java:129]: praise: nice, thanks for checking this case as well

**Comment by bion** [comment]
Looks good! Ready to approve, looks like it just needs formatting. You can run the auto formatter with `./bin/fmt`from the root directory.

**Review by bion — APPROVED** [review]
